Pr vision cam

EliaTarasov opened the pull request:
This pr adds vision cam model to be able to test external vision without internal vision plugin (for example by using ROS).

EliaTarasov committed:
add vision cam for ros external vision

EliaTarasov committed:
add vision cam for ros external vision

LorenzMeier committed:
Gazebo: Fix image trigger feedback

EliaTarasov committed:
add iris_vision model explicitly

EliaTarasov committed:
add iris_vision model explicitly

EliaTarasov committed:
change magnetic field tag values to pass travis parser

EliaTarasov committed:
change magnetic field tag values to pass travis parser

TSC21 commented on models/vision_cam/model.sdf:
```diff
@@ -0,0 +1,60 @@
+            <stddev>0.001</stddev>
+          </noise>
+        </camera>
+        <plugin name="svo_plugin" filename="libgazebo_ros_camera.so">
```
`svo_plugin`?

EliaTarasov replied:
I've tested this model with external vision library which is called svo: https://github.com/uzh-rpg/rpg_svo. But it's up to user what name for plugin is preffered. We may call it vision_plugin.

EliaTarasov committed:
choose consistent name for plugin

EliaTarasov committed:
choose consistent name for plugin

ChristophTobler committed:
change range finder topic to be independent of model

TSC21 commented on models/vision_cam/model.sdf:
```diff
@@ -0,0 +1,60 @@
+          </noise>
+        </camera>
+        <plugin name="vision_cam_plugin" filename="libgazebo_ros_camera.so">
+	  <alwaysOn>true</alwaysOn>
```
The indentation is 2 whitespaces.

EliaTarasov committed:
fix indentation

EliaTarasov committed:
fix indentation

EliaTarasov committed:
Update model.sdf

EliaTarasov committed:
Update model.sdf

RomanBapst committed:
mavlink_interface: added flag to indicate tailsitter

- since the airspeed axis is different for tailsitters we need to know
if the vehicle is a tailsitter or not

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
mavlink_interface.h: added some missing initialisations

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
tailsitter.sdf: set vehicle_is_tailsitter flag

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
updated xacro template to include vehicle_is_tailsitter flag

Signed-off-by: Roman <bapstroman@gmail.com>

dogmaphobic committed:
Basic Camera Control (#181)

This implements the Basic Camera Operations described by the Camera Protocol Documentation:
https://mavlink.io/en/protocol/camera.html

Only the very basic commands are implemented, providing image capture only (no video capture). Within the basic set of messages, all messages are implemented as well as all behaviors.

This "camera" component handles its own UDP "link" with the GCS. It emits its own heartbeats, and handles all the camera related communication between the GCS and the camera.

julianoes committed:
travis: exit on failure using bash -e

julianoes committed:
gazebo_geotagged_images_plugin: fix build error

RomanBapst committed:
added a model for a quad tiltrotor

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
gazebo_mavlink_interface: removed lots of obsolete code

- handling of the various actuation joints is handled via
channels specified in the sdf file of the model

Signed-off-by: Roman <bapstroman@gmail.com>

ChristophTobler committed:
lidar plugin: fix ambiguity problem for isinf() when using c++11

RomanBapst committed:
tiltrotor model: reduce rotor drag coefficient:

- this was causing unrealistically high side force when the vehicle was turning
and producing sideslip, as a result the vehicle would hardly turn

Signed-off-by: Roman <bapstroman@gmail.com>

EliaTarasov committed:
Merge branch 'pr-vision_cam' of https://github.com/EliaTarasov/sitl_gazebo into pr-vision_cam
